### ISSUE TITLE
Problem with mingw32-g++ 6.3.0 triggered by std::ifstream constructor?

### ISSUE THREAD

**Issue by dnso86** [cnjinhao/nana#481]
I believe this line triggers a weird problem in `mingw32-g++ 6.3.0` and possibly in other versions too:

https://github.com/cnjinhao/nana/blob/3cb7dfae0245d364bc46851195622727b2c68aeb/source/paint/image.cpp#L173

it is related in a way or another to these vague reports [1](https://gcc.gnu.org/bugzilla/show_bug.cgi?format=multiple&id=48547), [2](https://stackoverflow.com/questions/34124994/is-this-simple-c-program-using-locale-correct), [3](https://trac.filezilla-project.org/ticket/5168). 

Calling `form.icon()` leads there - and in the debugger it looks like the copy constructor of `std::locale` is called on some rubbish on the stack, despite the binary mode being specified, where locale wouldn't be used. Might be related to some name-mangling / ABI changes but as turning off optimizations is a quick fix I did not look into it further.

The bottom line is: it is not an issue in **nana** at all. 

However, what do you think, should we document this behaviour - besides this issue - in a comment somewhere in the source too?

**Comment by qPCR4vir** [comment]
try ` std::ifstream ifs(p.string(), std::ios::binary); `?

**Comment by dnso86** [comment]
Hi @qPCR4vir - I totally forgot this one. As far as I recall, it turned out to be some mixup regarding the ABIs in the build environment. Closing it, there is (was) nothing to do with this in nana :)
